libr/util/log.c

```diff
@@ -31,3 +31,33 @@ R_API void r_log_error(const char *str) {
 R_API void r_log_progress(const char *str, int percent) {
 	printf ("%d%%: %s\n", percent, str);
 }
+
+R_API void r_vlogf(RLogLevel level, const char *fmt, va_list ap) {
+	static const char *headers[R_LOG_MAX_VALUE] = {
+		[R_LOG_DEBUG] = "DEBUG: ",
+		[R_LOG_INFO] = "INFO: ",
+		[R_LOG_WARNING] = "WARNING: ",
+		[R_LOG_ERROR] = "ERROR: ",
+		[R_LOG_CRITICAL] = "CRITICAL: ",
+	};
+
+	const char *hdr = R_BETWEEN (0, level, R_LOG_MAX_VALUE - 1) ? headers[level] : "";
+	size_t len = strlen (hdr);
+	char buf[LOG_BUF_SZ];
+
+	snprintf (buf, LOG_BUF_SZ, "%s", hdr);
```

Comment:
i prefer sizeof(buf) rather than reusing the define, because the define doesn't needs to be always tied to the buffer size. it's described in the CERT secure C coding guidelines as a recommendation.